mne_icalabel/features/topomap.py

```diff
@@ -0,0 +1,121 @@
+        names or indices are explicitly provided.
+
+    Returns
+    -------
+    topo_array : np.ndarray of shape (n_components, n_pixels, n_pixels)
+    """
+    n_components = ica.mixing_matrix_.shape[1]
+    data = np.dot(
+        ica.mixing_matrix_[:, : ica.n_components_].T,
+        ica.pca_components_[: ica.n_components_],
+    )
+    # Create an empty array of size (n_components, 64, 64) to fit topo values
+    topo_array = np.zeros((n_components, 64, 64))
+
+    # f, ax = plt.subplots(1, ica.n_components_) #For Visualization
+    for j in range(n_components):
+        topo_ = np.flipud(topographic_map(data[j, :], ica.info))
+
+        # ax[j].imshow(topo_) # For visualization
+
+        # Set NaN values to 0
+        np.nan_to_num(topo_, copy=False)
+
+        # Standardize the values
+        topo_array[j, :, :] = topo_ / np.max(np.abs(topo_))
+
+    return topo_array  # topographic map array for all the components (n_components, 64, 64)
+
+
+def topographic_map(
+    data,
+    pos: Info,
+    vmin=None,
+    vmax=None,
+    res: int = 64,
+    outlines="head",
+    image_interp=_INTERPOLATION_DEFAULT,  #'cubic
+    border=_BORDER_DEFAULT,  #'mean'
+    extrapolate=_EXTRAPOLATE_DEFAULT,  #'head'
+):
+    """Generate a topographic map as image.
+
+    Parameters
+    ----------
+    data : array of shape (n_channels,)
+        The data values to plot.
+    pos : instance of `mne.Info`
+    vmin : float | callable | None
+        The value specifying the lower bound of the color range.
+    vmax : float | callable | None
+        The value specifying the upper bound of the color range.
+    res : int = 64
+        The resolution of the square topomap image (in pixels).
+    %(outlines_topomap)s
+    image_interp : str
+        The image interpolation to be used. All matplotlib options are
+        accepted.
+    %(border_topomap)s
+    %(extrapolate_topomap)s
+
+    Returns
+    -------
+    Zi : Array of size (n_pixels, n_pixels)
+        Topographic map array
+    """
+    picks = _pick_data_channels(pos, exclude=())  # pick only data channels
+    pos = pick_info(pos, picks)
+    ch_type = _get_channel_types(pos, unique=True)
+
+    if len(ch_type) > 1:
+        raise ValueError("Multiple channel types in Info structure.")
+    elif len(pos["chs"]) != data.shape[0]:
+        raise ValueError("Number of channels in the Info object the data array do not match.")
+    else:
+        ch_type = ch_type.pop()
+
+    picks = list(range(data.shape[0]))
+    sphere = np.array([0.0, 0.0, 0.0, 0.095])
+
+    # inferring x,y coordinates form mne.info instance
+    pos = _find_topomap_coords(pos, picks=picks, sphere=sphere)
+    extrapolate = _check_extrapolate(extrapolate, ch_type)
+
+    if len(data) != len(pos):
+        raise ValueError(
+            "Data and pos need to be of same length. Got data of "
+            "length %s, pos of length %s" % (len(data), len(pos))
+        )
```

Comment:
This check looks like a duplicate of the one above (ValueError: The number of channels ...), or am I missing something?